derive_builder/src/options/struct_options.rs

```diff
@@ -52,6 +56,20 @@ impl StructOptions {
             initializers: Vec::with_capacity(self.struct_size_hint),
             doc_comment: None,
             no_std: self.no_std,
+            struct_default: if cfg!(feature = "struct_default") {
+                self.struct_default.as_ref().map(struct_default)
+            } else {
+                None
+            },
         }
     }
 }
+
+/// Converts a DefaultExpression into the correct code block for the initializer.
+fn struct_default(expr: &DefaultExpression) -> Block {
+    (match *expr {
+        DefaultExpression::Struct 
+        | DefaultExpression::Trait => "::std::default::Default::default()".parse(),
+        DefaultExpression::Explicit(ref body) => format!("{}()", body).parse()
```

Comment:
Flagging this for closer scrutiny, as it touches on the issue described in #65 (this change is just at the struct level; the field level can be a separate PR).